include/deal.II/lac/precondition.h

```diff
@@ -476,6 +476,29 @@ class PreconditionJacobi : public PreconditionRelaxation<MatrixType>
 {
 public:
   /**
+   * AdditionalData to store relaxation parameter.
+   */
+  struct AdditionalData
```

Comment:
i would still need to have a non-default constructor of `AdditionalData ` with relaxation parameter and call `PreconditionRelaxation::AdditionalData` from within. Not sure it is really worth it. But if you insist, i can switch to this scheme.